### PR TITLE
Add login feature and jwt checks for authenticated endpoint

### PR DESCRIPTION
**Pull request by narutox900**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/alitonia/multi-choice-system/issues/81

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

This PR introduces authentication using JWT token. From now on, authenticated router will now require valid access token in order to work. We can now access the account_id of the user with 
`principal: Principal = Depends(security.get_current_user)`
used inside the api params.

### CODE REVIEW

**Comment by narutox900** [comment]
We don't need logout endpoint since we can do that on the client side.
The logout endpoint is only needed if the login system is implemented more complex with sessions and such.

**Comment by alitonia** [comment]
There are some errors with importing when running `docker-compose` from repository root. Can you double check this?

**Comment by narutox900** [comment]
> There are some errors with importing when running `docker-compose` from repository root. Can you double check this?

Pydantic model shouldn't contain too much information about SQLalchemy model I think. I have temporarily removed the relationship reference in the Pydantic model. Please check this.

**Comment by alitonia** [comment]
I fix an issue with login, but didn't test everything. Please do not merge this yet 🤠

File: backend/src/app/services/account_service.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ async def get_one_account(self, account_id: int):
 
     async def get_account_by_email(self, email: str) -> Optional[Account]:
         result = await self.session.execute(select(Account).where(Account.email == email))
-        return result.scalar().first()
+        return result.scalar()
```
Comment (narutox900) [inline backend/src/app/services/account_service.py:72]: Why did you remove the first() here? 
first() will return the first object in the query.

Comment (alitonia) [reply backend/src/app/services/account_service.py:72]: When I run the code and tries to login, it errors with cannot find method first(). printing result.scalar() already shows an Account object, so I find no need to call first(). Do you encounter the same problem as I did?

Comment (narutox900) [reply backend/src/app/services/account_service.py:72]: Oh my previous code should have been `scalars`. 
My bad, thank you.

**Review by alitonia — APPROVED** [review]
Fixed some problems with url resolution ( account/{account_id} and account/current ; and implicit encode/decode jwt. Need someone to run the code to confirm before commend merging procedure.
